tools/payments/cmd/chain-account/main.go

```diff
@@ -0,0 +1,238 @@
+
+Usage:
+
+chain-account [flags] generatePUBLIC_KEY
+
+The flags are:
+
+	-pr
+		Location on file system of the original prepared report
+	-v
+		verbose logging enabled
+	-k
+		Location on file system of the operators private ED25519 signing key in PEM format.
+	-e
+		The environment to which the operator is sending approval for transactions.
+*/
+package main
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"crypto/ed25519"
+	"encoding/hex"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	client "github.com/brave-intl/bat-go/libs/clients"
+	"github.com/brave-intl/bat-go/libs/httpsignature"
+	"github.com/brave-intl/bat-go/libs/payments"
+	paymentscli "github.com/brave-intl/bat-go/tools/payments"
+)
+
+func main() {
+	ctx := context.Background()
+
+	// command line flags
+	operatorKey := flag.String(
+		"k", "test/private.pem",
+		"the operator's key file location (ed25519 private key) in PEM format")
+
+	env := flag.String(
+		"e", "local",
+		"the environment to which the tool will interact")
+
+	verbose := flag.Bool(
+		"v", false,
+		"view verbose logging")
+
+	pcr2 := flag.String(
+		"pcr2", "", "the hex PCR2 value for this enclave")
+
+	flag.Parse()
+
+	args := flag.Args()
+	if len(args) < 2 {
+		log.Fatal("Expected chain id and action arguments")
+	}
+
+	switch args[0] {
+	case "solana":
+		switch args[1] {
+		case "generate":
+			generateSolanaAddress(ctx, *operatorKey, *env, *pcr2, *verbose)
+		case "approve":
+			if len(args) < 3 {
+				log.Fatal("Expected public key argument for approval")
+			}
+			pubKey := args[2]
+			approveSolanaAddress(ctx, pubKey, *operatorKey, *env, *pcr2, *verbose)
+		default:
+			log.Fatal("unrecognized solana command")
+		}
+	default:
+		log.Fatal("unrecognized chain id")
+	}
+
+}
+
+func generateSolanaAddress(ctx context.Context, key, env, pcr2 string, verbose bool) {
+	if env != "development" && len(pcr2) != 96 {
+		log.Fatal("a valid pcr2 is required to generate an address outside of development\n")
+	}
+
+	priv, err := paymentscli.GetOperatorPrivateKey(key)
+	if err != nil {
+		log.Fatalf("failed to parse operator key file: %v\n", err)
+	}
+	var (
+		// dateHeader needs to be lowercase to pass the signing verifier validation.
+		dateHeader          = "date"
+		contentLengthHeader = "Content-Length"
+		contentTypeHeader   = "Content-Type"
+	)
+
+	signer := httpsignature.ParameterizedSignator{
+		SignatureParams: httpsignature.SignatureParams{
+			Algorithm: httpsignature.ED25519,
+			KeyID:     hex.EncodeToString([]byte(priv.Public().(ed25519.PublicKey))),
+			Headers: []string{
+				"(request-target)",
+				"date",
+				"digest",
+				"content-length",
+				"content-type",
+			},
+		},
+		Signator: priv,
+		Opts:     crypto.Hash(0),
+	}
+
+	buf := bytes.NewBuffer([]byte{})
+	buf.WriteString(key)
+	body := buf.Bytes()
+	if err != nil {
+		log.Fatalf("failed to marshal attested transaction body: %w", err)
+	}
+	apiBase := os.Getenv("NITRO_API_BASE")
```

Comment:
We should add into the docs what this should be set to and that it's necessary to run the CLI